onnxruntime/core/providers/cpu/math/clip.cc

```diff
@@ -70,16 +70,37 @@ template <typename T>
 Status Clip_6<T>::Compute(OpKernelContext* ctx) const {
   const auto* X = ctx->Input<Tensor>(0);
   Tensor* Y = ctx->Output(0, X->Shape());
-  EigenVectorMap<T>(Y->MutableData<T>(), onnxruntime::narrow<size_t>(Y->Shape().Size()) ) =
-      ConstEigenVectorMap<T>(X->Data<T>(), onnxruntime::narrow<size_t>(X->Shape().Size()) )
-          .cwiseMax(this->min_)
-          .cwiseMin(this->max_);
+
+  // We split input/output data into chunks. Each chunk has N elements 
+  // (except, maybe, the last chunk), and we use a thread pool to process
+  // the chunks in parallel. N = 16384 was selected based on performance
+  // results on input tensors of 10^i elements for i in [1 .. 6].
+  static constexpr int64_t length_per_task = 16384;
+  
+  int64_t elem_count = Y->Shape().Size();
+  int64_t task_count = (elem_count + length_per_task - 1) / length_per_task;
+
+  concurrency::ThreadPool::TryBatchParallelFor(
```

Comment:
> concurrency::ThreadPool::TryBatchParallelFor( [](http://example.com/codeflow?start=1&length=46)

Suggest lambda as a local var and for a single chunk not to invoke any of the Threadpool code as it may instantiate lambdas that allocate memory.

Reply:
Just making sure I understand your comment -- are you suggesting doing something like
`auto clip = [...] {...};`

`// Parallelize only when we have more than one task`
`if (task_count > 1) TryBatchParallelFor(tp, task_count, clip, 0);`
`else clip(0);`

If so, TryBatchParallelFor already has precisely [that logic in place](https://github.com/microsoft/onnxruntime/blob/acbb7ad45335aa1721cad0180e94d626a0d45cb6/include/onnxruntime/core/platform/threadpool.h#L295).

Reply:
Ok